consensus/scp/tests/mock_network/mod.rs

```diff
@@ -25,6 +20,66 @@ use std::{
     time::{Duration, Instant},
 };
 
+// Controls test parameters
+#[derive(Clone)]
+pub struct TestOptions {
+    /// Values can be submitted to all nodes in parallel (true) or to nodes in sequential order (false)
+    pub submit_in_parallel: bool,
+
+    /// Total number of values to submit. Tests run until all values are externalized by all nodes.
+    pub values_to_submit: usize,
+
+    /// Approximate rate that values are submitted to nodes.
+    pub submissions_per_sec: u64,
+
+    /// We allow only a single proposal per slot, with up to this many values.
+    pub max_values_per_slot: usize,
+
+    /// The total allowed testing time before forcing a panic
+    pub allowed_test_time: Duration,
+
+    /// wait this long for slog to flush values before ending a test
+    pub log_flush_delay: Duration,
+
+    /// This parameter sets the interval for round and ballot timeout.
+    /// SCP suggests one second, but threads can run much faster.
+    pub scp_timebase: Duration,
+
+    /// Sleep when advancing to a new slot to help threads keep pace.
+    pub slot_advance_delay: Duration,
+
+    /// Because thread testing doesn't implement catchup, increase the lrucache used to store msgs.
+    pub seen_msg_hashes_lru_size: usize,
```

Comment:
I still don't think this is necessary. The seen messages LRU only prevents processing of messages we've already processed.